speechbrain/nnet/RNN.py

```diff
@@ -150,7 +151,11 @@ def forward(self, x, hx=None, lengths=None):
         self.rnn.flatten_parameters()
 
         # Pack sequence for proper RNN handling of padding
-        if lengths is not None:
+        if (
+            lengths is not None
+            and not self.training
```

Comment:
I'm not sure about this line. This seems to lead to something different from the previous code as we call   `pack_padded_sequence` only when we are in eval modality.

Reply:
you were right, I fixed it in [2e44f97](https://github.com/speechbrain/speechbrain/pull/1738/commits/2e44f978f23b76e2b939b00c2ff1975436708865)